File: GooglePay/src/main/java/com/braintreepayments/api/googlepay/GooglePayPaymentAuthRequestParams.java
package com.braintreepayments.api.googlepay;

import androidx.annotation.NonNull;

import com.google.android.gms.wallet.PaymentDataRequest;

/**
 * Used to request Google Pay payment authorization via
 * {@link GooglePayLauncher#launch(GooglePayPaymentAuthRequestParams)}
 */
public class GooglePayPaymentAuthRequestParams {

    private final int googlePayEnvironment;
    private final PaymentDataRequest paymentDataRequest;

    GooglePayPaymentAuthRequestParams(int googlePayEnvironment, @NonNull PaymentDataRequest paymentDataRequest) {
        this.googlePayEnvironment = googlePayEnvironment;
        this.paymentDataRequest = paymentDataRequest;
    }

    int getGooglePayEnvironment() {
        return googlePayEnvironment;
    }

    PaymentDataRequest getPaymentDataRequest() {
        return paymentDataRequest;
    }
}

File: GooglePay/src/main/java/com/braintreepayments/api/googlepay/GooglePayPaymentAuthRequestParams.kt
package com.braintreepayments.api.googlepay

import androidx.annotation.RestrictTo
import com.google.android.gms.wallet.PaymentDataRequest

/**
 * Used to request Google Pay payment authorization via
 * [GooglePayLauncher.launch]
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
class GooglePayPaymentAuthRequestParams internal constructor(
    val googlePayEnvironment: Int,
    val paymentDataRequest: PaymentDataRequest
)
